sdk/transport_policies/curl/src/az_curl.c

```diff
@@ -79,8 +79,8 @@ az_span const AZ_HTTP_REQUEST_BUILDER_HEADER_SEPARATOR = AZ_SPAN_LITERAL_FROM_ST
  * @param separator symbol to be used between key and value
  * @return az_result
  */
-AZ_NODISCARD az_result
-az_write_to_buffer(az_span writable_buffer, az_pair header, az_span separator) {
+static AZ_NODISCARD az_result
+_az_span_write_to_buffer(az_span writable_buffer, az_pair header, az_span separator) {
```

Comment:
Can we rename this to _az_span_append_header_to_buffer because it calls append internally and this indicates what the function actually appends to the span